### ISSUE TITLE
Determine approach for multiple repositories

### ISSUE THREAD

**Issue by mchelen-gov** [department-of-veterans-affairs/va.gov-team#14815]
## Background
Projects such as va.gov involve multiple repositories for development. See for full list

## Goal
Make it as easy as possible to work on projects using multiple repositories, and also projects that involve only 1 repository

### Config-per-repo
 - Each code repo (`vets-website`, `vets-api`, etc) has its own Codespace `.devcontainer` config
 - The repo's config automatically clones *direct dependencies* and nothing else
 - For example `vets-website` directly requires the content repo, so this is [automatically cloned](https://github.com/department-of-veterans-affairs/vets-website/blob/master/.devcontainer/post-create.sh#L20)
 - Other repositories such as `vets-api` which are not *directly required* by `vets-website` would not automatically be cloned, and would be up to the user to install as needed

### Centralized config
- All Codespaces `.devcontainer` config is placed in a single [central repo](https://github.com/department-of-veterans-affairs/codespaces)
- Each code repo (`vets-website`, `vets-api`, etc) does not contain any Codespaces config
- The central config automatically clones and configures all code repositories (`vets-website`, `vets-api`, etc) 

### Config inheritance?
- Common config is placed in a central location
- Each code repo inherits and extends this config with repo-specific customizations

**Comment by mchelen-gov** [comment]
@Chuxel does this look like an approximate summary of options for projects with multiple repositories?
I haven't dug into the pros/cons for each yet.

**Comment by Chuxel** [comment]
@mchelen-gov Yep! If needed, you could also have scenario based central config. e.g. a separate "bootstrap" repo for each type or style of development.
